substrate/executor/src/native_executor.rs

```diff
@@ -203,15 +203,15 @@ macro_rules! native_executor_instance {
 	};
 	(IMPL $name:ident, $dispatcher:path, $version:path, $code:expr) => {
 		// TODO: this is not so great – I think I should go back to have dispatch take a type param and modify this macro to accept a type param and then pass it in from the test-client instead
-		use primitives::KeccakHasher as _KeccakHasher;
+		use primitives::Blake2Hasher as _Blake2Hasher;
```

Comment:
@guanqun's recent [PR](https://github.com/paritytech/substrate/pull/683) originally had a fix for this terrible hack now lost after re-working it; perhaps we can sort that out in this PR too?

Reply:
second PR would be good if @guanqun fancies putting it in.